src/odb/src/zutil/util.cpp

```diff
@@ -130,6 +130,32 @@ void RUDYCalculator::calculateRUDY()
       processMacroObstruction(master, instance);
     }
   }
+
+  float rudy_adjustment = 0.0;
+  odb::dbTech* tech = block_->getTech();
+  for (int l = 1; l <= tech->getRoutingLayerCount(); l++) {
+    odb::dbTechLayer* tech_layer = tech->findRoutingLayer(l);
+    if (tech_layer == nullptr) {
+      continue;
+    }
+    rudy_adjustment += tech_layer->getLayerAdjustment();
```

Comment:
Averaging the adjustments isn't precise.  The number of tracks varies by layer so the effect of a given derate is larger on layers with more tracks.

Reply:
Alright. What do you think of using the greatest value then?

Reply:
A pitch weighted average would be closer (though still subject to some round-off).

Reply:
The idea is that layers with higher pitch values should weight more on the derating average for RUDY, correct?

Reply:
lower pitch implies more tracks so you want to more weight on lower pitch

Reply:
You can check the behavior for [three methods on this link](https://drive.google.com/drive/folders/1T-ECrPp1tulA_uU4xfXOWSH5nAK45KcJ?usp=sharing):  1. Using the maximum derating value across all layers, 2. Calculating a simple average of the derating values across layers, and 3. the current implementation on this PR, with a weighted average based on the pitch value of each layer.

Reply:
You can see [on this link](https://docs.google.com/spreadsheets/d/1FpQRPGZ4ImhqMTdEhdUEsyUx0nXm6fZtQTUh8IQDWOg/edit#gid=0) that the correlation metrics (RUDY versus GRT) are quite similar. Their differences are more evident on the heatmap images.